internal/expr/typeinfo.go

```diff
@@ -70,14 +84,17 @@ type structInfo struct {
 	// Ordered list of tags
 	tags []string
 
-	tagToField map[string]*structField
+	tagToField map[string]structField
 }
 
 func (si *structInfo) typ() reflect.Type {
 	return si.structType
 }
 
 func (si *structInfo) typeMember(member string) (typeMember, error) {
+	if member == "" {
+		return nil, fmt.Errorf(`type %q missing struct db tag`, si.structType.Name())
+	}
```

Comment:
The fact that this error has to be reported here once we have done all the validation and type checking tells me it should really be move to the parser.

Reply:
It would be great to but at the parse stage we don't know if an unqualified named is a named primitive or an actual unqualified map/struct.